test/index.js

```diff
@@ -1,472 +1,112 @@
-  private: {
-    key: fs.readFileSync(__dirname + '/pass.rsa.2028.priv'),
-    passphrase: 'password'
-  },
-  public: fs.readFileSync(__dirname + '/pass.rsa.2028.pub')
-}
-var vector = fs.readFileSync(__dirname + '/vector.priv')
-var vector2 = fs.readFileSync(__dirname + '/vector2.priv')
 function isNode10 () {
   return process.version && process.version.split('.').length === 3 && parseInt(process.version.split('.')[1], 10) <= 10
 }
-var nodeCrypto = require('crypto')
-var myCrypto = require('../browser')
-function testIt (keys, message, scheme) {
-  var pub = keys.public
-  var priv = keys.private
+
+fixtures.valid.rsa.forEach(function(f) {
+  var message = new Buffer(f.message)
+  var pub = new Buffer(f.public, 'base64')
+  var priv
+
+  if (f.passphrase) {
+    priv = {
+      key: new Buffer(f.private, 'base64'),
+      passphrase: f.passphrase
+    }
+  } else {
+    priv = new Buffer(f.private, 'base64')
+  }
+
   test(message.toString(), function (t) {
-    t.plan(4)
-    var mySign = myCrypto.createSign(scheme)
-    var nodeSign = nodeCrypto.createSign(scheme)
+    t.plan(5)
+
+    var mySign = myCrypto.createSign(f.scheme)
+    var nodeSign = nodeCrypto.createSign(f.scheme)
     var mySig = mySign.update(message).sign(priv)
     var nodeSig = nodeSign.update(message).sign(priv)
+
     t.equals(mySig.length, nodeSig.length, 'correct length')
     t.equals(mySig.toString('hex'), nodeSig.toString('hex'), 'equal sigs')
-    var myVer = myCrypto.createVerify(scheme)
-    var nodeVer = nodeCrypto.createVerify(scheme)
+    t.equals(mySig.toString('hex'), f.signature)
+
+    var myVer = myCrypto.createVerify(f.scheme)
+    var nodeVer = nodeCrypto.createVerify(f.scheme)
     t.ok(nodeVer.update(message).verify(pub, mySig), 'node validate my sig')
     t.ok(myVer.update(message).verify(pub, nodeSig), 'me validate node sig')
   })
-}
-function ectestIt (keys, message, scheme) {
-  var pub = keys.public
-  var priv = keys.private
+})
+
+fixtures.valid.ec.forEach(function(f) {
+  var message = new Buffer(f.message)
+  var pub = new Buffer(f.public, 'base64')
+  var priv
+
+  if (f.passphrase) {
+    priv = {
+      key: new Buffer(f.private, 'base64'),
+      passphrase: f.passphrase
+    }
+  } else {
+    priv = new Buffer(f.private, 'base64')
+  }
+
+  // skip node 10 tests
+  if (f.node11 && isNode10()) return
+
   test(message.toString(), function (t) {
     t.plan(3)
 
-    var nodeSign = nodeCrypto.createSign(scheme)
-    var mySign = myCrypto.createSign(scheme)
+    var nodeSign = nodeCrypto.createSign(f.scheme)
+    var mySign = myCrypto.createSign(f.scheme)
 
     var mySig = mySign.update(message).sign(priv)
     var nodeSig = nodeSign.update(message).sign(priv)
     t.notEqual(mySig.toString('hex'), nodeSig.toString('hex'), 'not equal sigs')
-    var myVer = myCrypto.createVerify(scheme)
-    var nodeVer = nodeCrypto.createVerify(scheme)
+
+    // FIXME: deterministic signatures?
+//    t.equals(mySig.toString('hex'), f.signature)
```

Comment:
Wait ecdsa might not be deterministic its handled by elliptical

On Wed, Mar 11, 2015, 5:45 AM Calvin Metcalf <calvin.metcalf@gmail.com>
wrote:

> How did you calculate it?
>
> On Tue, Mar 10, 2015, 10:08 PM Daniel Cousens <notifications@github.com>
> wrote:
>
>> In test/index.js
>> <https://github.com/crypto-browserify/browserify-sign/pull/11#discussion_r26184255>
>> :
>>
>> >
>> >      var mySig = mySign.update(message).sign(priv)
>> >      var nodeSig = nodeSign.update(message).sign(
>>
>> priv)
>> >      t.notEqual(mySig.toString('hex'), nodeSig.toString('hex'), 'not equal sigs')
>> > -    var myVer = myCrypto.createVerify(scheme)
>> > -    var nodeVer = nodeCrypto.createVerify(scheme)
>> > +
>> > +    // FIXME: deterministic signatures?
>> > +//    t.equals(mySig.toString('hex'), f.signature)
>>
>> Then why does uncommenting that line make the tests fail?
>> … <#msg-f:1495339755103849914_msg-f:1495311001811438367_>
>> On 11 Mar 2015 11:55 am, "Calvin Metcalf" <notifications@github.com>
>> wrote: In test/index.js <https://github.com/crypto-
>> browserify/browserify-sign/pull/11#discussion_r26181654> : > > var mySig
>> = mySign.update(message).sign(priv) > var nodeSig =
>> nodeSign.update(message).sign(priv) > t.notEqual(mySig.toString('hex'),
>> nodeSig.toString('hex'), 'not equal sigs') > - var myVer =
>> myCrypto.createVerify(scheme) > - var nodeVer = nodeCrypto.createVerify(scheme)
>> > + > + // FIXME: deterministic signatures? > +//
>> t.equals(mySig.toString('hex'), f.signature) Our ecdsa ones are
>> deterministic but node's are not — Reply to this email directly or view it
>> on GitHub <https://github.com/crypto-browserify/browserify-sign/
>> pull/11/files#r26181654> .
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/crypto-browserify/browserify-sign/pull/11/files#r26184255>
>> .
>>
>